src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Storage/IPluginsStorageDirectory.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.Performance.Toolkit.Plugins.Core;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Installation
+{
+    /// <summary>
+    ///     Represents the structure of the directory on a file system where installed plugins are stored or should be stored.
+    ///     All installed plugins of a given plugins system should be stored under the same root directory.
+    /// </summary>
+    public interface IPluginsStorageDirectory
+    {
+        /// <summary>
+        ///     Gets the dedicated directory where the given plugin is installed or should be installed.
+        /// </summary>
+        /// <param name="pluginIdentity">
+        ///     The identity of the plugin.
+        /// </param>
+        /// <returns>
+        ///     The directory where the given plugin is installed or should be installed.
```

Comment:
I wonder if we want these methods to be returning _relative_ paths. That way the implementation doesn't need to prepend some "root directory," but rather the consumers do. Thoughts?

Reply:
Returning relative paths does make it more flexible but this interface has a lot of consumers (storage, loader, remover, checksum calculator and validator) and all of them will need to know about the plugins system root and append relative paths if we make this change. And these components are supposed to be sharing one root directory. Would it be better to just have `IPluginsStorageDirectory` to store the root?

Reply:
That's a good point. I'd say leave it as-is and have this class store the root